File: programmers/js/level00/120878.js
// 유한소수 판별하기
function solution(a, b) {
  const FINITE_DECIMAL = 1;
  const INFINITE_DECIMAL = 2;
  const denominator = b / gcd(a, b);

  return isFiniteDecimal(denominator) ? FINITE_DECIMAL : INFINITE_DECIMAL;
}

function gcd(a, b) {
  while (b !== 0) {
    let tmp = a;
    a = b;
    b = tmp % a;
  }
  
  return a;
}

function isFiniteDecimal(num) {
  while (true) {
    if (num === 1) {
      return true;
    } else if (num % 2 === 0) {
      num /= 2;
    } else if (num % 5 === 0) {
      num /= 5;
    } else {
      return false;
    }
  }
}

// 다른 풀이
// function solution(a, b) {
//   return Number((a / b).toFixed(10)) == a / b ? 1 : 2;
// }

File: programmers/java/level00/120878.java
// 유한소수 판별하기

class Solution {
    int gcd(int a, int b) {
        while (b != 0) {
            int r = a;
            a = b;
            b = r % b;
        }
        
        return a;
    }
    
    boolean isFiniteDecimal(int num) {
        while (true) {
            if (num == 1) {
                return true;
            } else if (num % 2 == 0) {
                num /= 2;
            } else if (num % 5 == 0) {
                num /= 5;
            } else {
                return false;
            }
        }
    }
    
    public int solution(int a, int b) {
        int denominator = b / gcd(a, b);
        
        return isFiniteDecimal(denominator) ? 1 : 2;
    }
}
